include/swift/AST/DiagnosticsSema.def

```diff
@@ -2451,6 +2465,12 @@ REMARK(add_predates_concurrency_import,none,
      "%select{| as warnings}0", (bool, Identifier))
 REMARK(remove_predates_concurrency_import,none,
      "'@preconcurrency' attribute on module %0 is unused", (Identifier))
+WARNING(remove_public_import,none,
+        "public import of %0 was not used in public declarations or inlinable code",
```

Comment:
I think when diagnostics refer to inlinable code we usually name the `@inlinable` attribute directly to reduce ambiguity. It's a bit more verbose, but what about `in public declarations or @inlinable function bodies`?

Reply:
I believe this is the first diagnostic that mentions inlinable code being missing, so we don't have just one attribute we can reference. I went for "in public declarations or inlinable code" because I think it hits a right balance between too little and too much information. It hints at `@inlinable` while covering the 2-3 other attributes and the default argument value case. I also expect it to be often shown to people that don't use `@inlinable` and others, and where it may be more confusing than anything else.

An alternative would be to refer to `@inlinable` as a catch all or a family of attributes, it would avoid the ambiguity with implicitly inlinable code. 

If we want to go all the way, we could keep the general warning and add a note along the line of: "
public imports are required when referenced from public or `@usableFromInline` declarations signatures, `@inlinable` or `@backDeployed` function bodies, or default initializers of arguments and properties of `@frozen` struct."

What do you think?

Reply:
Yeah, we would usually refer to these contexts generically as "fragile" but we don't have a great word for this term to use in user-facing diagnostics. My thought process was that I was under the impression that `@inlinable` is already used as the representative of this concept in other diagnostics but now that I look again I can't find examples. FWIW, I don't think we really need to account for `@_alwasyEmitIntoClient` or `@backDeployed` in a diagnostic like this as they are features that are either unofficial or official but advanced - neither is encountered by most Swift users. A lot of Swift developers are going to see this diagnostic, though. I think I'm convinced that just `inlinable` is fine, but it might be worth a broader discussion with folks to align on terminology (it would be nice if there were a term officially defined in TSPL for this).

Reply:
How about "inline-eligible" or "inline-candidate"? 
e.g. "public import of Foo was not used in public declarations or inline-eligible code"

Reply:
We already have a clear definition of what is ABI. I'm thinking we might need something similar for what is part of the API of the module. Then we could reference that concept from diagnostics.